plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePartitionManager.java

```diff
@@ -130,7 +143,14 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         // All partition key domains will be fully evaluated, so we don't need to include those
         TupleDomain<ColumnHandle> remainingTupleDomain = effectivePredicate.filter((column, domain) -> !partitionColumns.contains(column));
         TupleDomain<ColumnHandle> enforcedTupleDomain = effectivePredicate.filter((column, domain) -> partitionColumns.contains(column));
-        return new HivePartitionResult(partitionColumns, partitionsIterable, compactEffectivePredicate, remainingTupleDomain, enforcedTupleDomain, hiveBucketHandle, bucketFilter);
+
+        /**
+         * Partitions will be parsed if
+         *  1. Number of partitionNames is less than or equal to threshold value.
+         *  2. If additional predicate is passed as a part of Constraint.
```

Comment:
Document why we're making this choice.

In any case, putting this boolean in `HivePartitionResult` is wrong.
The user of `HivePartitionResult` should do this logic (`getPartitionsAsList` or caller of it)
Remove `HivePartitionResult.canParsePartitions` fielld